be/src/http/action/stream_load.cpp

```diff
@@ -299,23 +307,40 @@ void StreamLoadAction::on_chunk_data(HttpRequest* req) {
     auto evbuf = evhttp_request_get_input_buffer(ev_req);
 
     int64_t start_read_data_time = MonotonicNanos();
-    while (evbuffer_get_length(evbuf) > 0) {
-        ByteBufferPtr bb = ByteBuffer::allocate(4096);
+
+    size_t len = 0;
+    while ((len = evbuffer_get_length(evbuf)) > 0) {
+        if (ctx->buf->remaining() < len) {
+            if (ctx->format == TFileFormatType::FORMAT_JSON) {
+                // For json format, we need build a complete json before we push the buffer to the pipe.
+                // buffer capacity is not enough, so we try to expand the buffer.
+                ByteBufferPtr buf = ByteBuffer::allocate(BitUtil::RoundUpToPowerOfTwo(ctx->buf->pos + len));
```

Comment:
```suggestion
                ByteBufferPtr buf = ByteBuffer::allocate(std::max(ctx->buf->pos + len, ctx->buf->capacity * 1.5)));
```

Reply:
This may slow down the expanding of buffer. What's your consideration?

Reply:
Reallocation in this way results in O(n^2) complexity for insertions.
Reference: https://github.com/facebook/folly/blob/main/folly/docs/FBVector.md